grpc-api/src/main/java/io/deephaven/grpc_api/appmode/ApplicationTicketResolver.java

```diff
@@ -51,66 +51,75 @@ public synchronized void onApplicationLoad(final ApplicationState app) {
     }
 
     @Override
-    public <T> SessionState.ExportObject<T> resolve(final @Nullable SessionState session, final ByteBuffer ticket) {
-        return resolve(appFieldIdFor(ticket));
+    public <T> SessionState.ExportObject<T> resolve(
+            final @Nullable SessionState session, final ByteBuffer ticket, final String logId) {
+        return resolve(appFieldIdFor(ticket, logId), logId);
     }
 
     @Override
-    public <T> SessionState.ExportObject<T> resolve(final @Nullable SessionState session,
-            final Flight.FlightDescriptor descriptor) {
-        return resolve(appFieldIdFor(descriptor));
+    public <T> SessionState.ExportObject<T> resolve(
+            final @Nullable SessionState session, final Flight.FlightDescriptor descriptor, final String logId) {
+        return resolve(appFieldIdFor(descriptor, logId), logId);
     }
 
-    private <T> SessionState.ExportObject<T> resolve(final AppFieldId id) {
+    private <T> SessionState.ExportObject<T> resolve(final AppFieldId id, final String logId) {
         if (id.app == null) {
-            throw new IllegalStateException("Cannot resolve field without knowing the application");
+            throw GrpcUtil.statusRuntimeException(Code.FAILED_PRECONDITION,
+                    "Could not resolve '" + logId + "': field '" + getLogNameFor(id)
+                            + "' does not belong to an application");
         }
         final Field<Object> field = id.app.getField(id.fieldName);
         if (field == null) {
-            throw GrpcUtil.statusRuntimeException(Code.NOT_FOUND, "ticket not found: " + getLogNameFor(id));
+            throw GrpcUtil.statusRuntimeException(Code.NOT_FOUND,
+                    "Could not resolve '" + logId + "': field '" + getLogNameFor(id) + "' not found");
         }
         // noinspection unchecked
         return SessionState.wrapAsExport((T) field.value());
     }
 
     @Override
-    public SessionState.ExportObject<Flight.FlightInfo> flightInfoFor(final @Nullable SessionState session,
-            final Flight.FlightDescriptor descriptor) {
-        final AppFieldId id = appFieldIdFor(descriptor);
+    public SessionState.ExportObject<Flight.FlightInfo> flightInfoFor(
+            final @Nullable SessionState session, final Flight.FlightDescriptor descriptor, final String logId) {
```

Comment:
wrong conv, but under what circumstances are we okay with session-less requests?

Reply:
We want to be able to expose any content through Flight. Possibly this feature should be gated by a flag and disabled by default.

Reply:
I added this comment to #997